api/src/LinkyLink/Service/LinksService.cs

```diff
@@ -35,7 +35,7 @@ public async Task<IEnumerable<LinkBundle>> AllLinkBundlesAsync()
         public async Task<LinkBundle> FindLinkBundleAsync(string vanityUrl)
         {
             return await _context.LinkBundle
-               .SingleAsync(b => b.VanityUrl == vanityUrl.ToLower());
+               .SingleOrDefaultAsync(b => b.VanityUrl == vanityUrl.ToLower());
```

Comment:
We need SingleOrDefaultAsync as this method returns empty if there is no record but throws exception when there is more than 1 record. SingleAsync throws exception when there is no record which is not a desired behavior as we use this method on Get and Delete on LinksController and we dont want to throws an exception when there is no record.

SingleOrDefaultAsync  should work for all scenarios as well as to support the scenario in challenge 2.